lib/pure/asyncdispatch.nim

```diff
@@ -1182,8 +1185,14 @@ else:
           else:
             retFuture.fail(newException(OSError, osErrorMsg(lastError)))
       else:
-        register(client.AsyncFD)
-        retFuture.complete((getAddrString(cast[ptr SockAddr](addr sockAddress)), client.AsyncFD))
+        try:
+          let address = getAddrString(cast[ptr SockAddr](addr sockAddress))
+          register(client.AsyncFD)
+          retFuture.complete((address, client.AsyncFD))
+        except:
+          # getAddrString may raise
```

Comment:
A general future remark, this worries me because it looks fragile.

Indeed, all procedures that return a Future shouldn't raise an exception but place the exception inside the Future.

Perhaps we should be using exception tracking to guarantee this?